### PR TITLE
Feature 1694 s2s subgroups

### LINKED ISSUE #947: Change all references from Convection Allowing Models to Short Range
Change parm/use_cases/model_applications/convection_allowing_models to parm/use_cases/model_applications/short_range and update all other references.

## Describe the Enhancement ##
This will involve many changes including use case paths, documentation, testing scripts, etc.

### Time Estimate ###
~1-2 days

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
2702691 2792541

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### LINKED ISSUE #1694: Internal: Add support for creating multiple input datasets for use case categories
Currently the automated tests are set up so that each model_applications category corresponds to an input data set that contains all of the data required to run all of the use cases in that category. The s2s input data set has become so large that while it doesn't exceed the maximum allowable size for the Docker data volume that stores it to use in the tests, but use case test groups that use this data run out of disk space when they write output data from the use cases.

The use case groups that fail also use the Conda environments required for METplotpy and METcalcpy, which are very large in size due to the many Python package dependencies. This also contributes to the total disk size that can be used in the test environment. The newly created metplotpy environment for #1566 is much larger in size than the existing environment, so this may cause disk space issues when that work is completed.

Size of current conda environments:
```
du -sh /usr/local/envs/*
897M    /usr/local/envs/metplotpy
185M    /usr/local/envs/metplus_base
```

Size of conda environments using Python 3.8.6 and updated package requirements:
```
du -sh /usr/local/envs/*
2.2G    /usr/local/envs/metplotpy.v5
168M    /usr/local/envs/metplus_base.v5
```

We may need to reconsider new requirements of use cases and how to group them in the tests, including:
* Size of input data
* Size out output data generated
* Size of conda environment required to run
* Others?

## Describe the Enhancement ##
- [ ] Come up with a good naming convention for the additional input data sets. Currently they are named after the category, i.e. s2s. We will need another data set such as s2s_2.
- [ ] Update the automated test logic to support multiple input data sets for a given category
- [ ] Update the Contributor's Guide Add Use Cases chapter with the updated process for adding new data
- [ ] Update User's Guide to describe how to find input data for use cases since they will not just correspond to the model_applications sub-directory name anymore

### Time Estimate ###
1-3 days

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
ASAP

### Funding Source ###
2702691 2792541

### Assignee ###
- [X] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [X] Select **component(s)**
- [X] Select **priority**
- [X] Select **requestor(s)**

### Projects and Milestone ###
- [X] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [X] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [X] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
### Change Summary
* Split s2s use cases into 3 groups (s2s, s2s_mid_lat, and s2s_mjo) to reduce the size of the input data sets to prevent GHA environment disk space issues
* Added logic recommended by GitHub developers to reduce disk size by removing files that are not needed by us, i.e. android dev tools
* Renamed convection_allowing_models category to short_range per #947 
* Modified logic to made repository rename from METdatadb to METdataio
* Changed exit code for diff tests so that it is easier to determine whether a use case group has failed due to errors in a use case (exit 1) or differences in the output (exit 2)

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Ensured all use cases ran successfully in automated tests

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

* Ensure all failures in automated tests are due to differences because the use case names and numbers have changed.
* Verify documentation of moved/renamed use cases are correct and formatted properly
* Check output files that flagged differences and ensure that the output is correct.
  * short_range use cases will correspond to the same number as the old convection_allowing_models equivalent
  * Most of the s2s use case numbers has shifted and some have been moved to new groups. The output should not change except that input/output directories that referenced s2s will now reference the new group names s2s_mid_lat and s2s_mjo for cases moved to those groups
* Optionally run use cases that have changed locally

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[Yes]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>
See above. Diffs will be shown in many use cases because they were renamed/renumbered. The actual output of these use cases should not differ.

- [X] Please complete this pull request review by **7/18/2022**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [X] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Contributors_Guide/add_use_case.rst [hunk 1]
```diff
@@ -42,7 +42,6 @@ one of the following:
 
 * air_quality_and_comp
 * climate
-* convection_allowing_models
 * data_assimilation
```
Comment (j-opatz) [inline docs/Contributors_Guide/add_use_case.rst:45]: Need to add in "short_range" to replace the "convection_allowing_models".

Comment (georgemccabe) [reply docs/Contributors_Guide/add_use_case.rst:45]: You are looking at the "before" text. This change is removing convection_allowing_models and adding short_range below.

**Review by CPKalb — APPROVED** [review]
Output from RMM, OMI, blocking, and weather regime match what they were prior to the change, so this looks good on my end.  We may want to uncover the settings to avoid overlapping documentation in the future

**Review by j-opatz — APPROVED** [review]
Change over for naming and new s2s categories looks to be all set. Documentation looks correct, and any missed references to convection allowing models at least shouldn't break the documentation.

**Pull request merged**